frontend/src/api/APIWorker.tsx

```diff
@@ -1,83 +1,79 @@
 import { useEffect, useRef, useState } from 'react'
```

Comment:
I know it's a refactoring, I think this should just be a pure function or maybe a class, no need for React here (especially if it's used as a webworker). And this is the power of Redux (vs `useReducer`), you can access it outside of functions (and dispatch its actions).

Maybe add a TODO for later on?